pkg/sql/drop_index.go

```diff
@@ -245,6 +245,13 @@ func (p *planner) dropIndexByName(
 		return nil
 	}
 
+	if idx.Unique && behavior != tree.DropCascade && constraintBehavior != ignoreIdxConstraint && !idx.CreatedExplicitly {
+		return pgerror.Newf(
+			pgcode.DependentObjectsStillExist,
+			"index %q is in use as unique constraint (use CASCADE if you really want to drop it)",
```

Comment:
i'd suggest maybe going a step further:

```
errors.WithHint(
  pgerror.Newf(
      pgcode.DependentObjectsStillExist,
			"index %q is in use as unique constraint",
   ),
   "Use CASCADE if you really want to drop it.",
)
```

that way the "error" and "remediation" are separate, but displays cleanly on UX. Happy to follow psql wording.